drivers/sensor/ina700/ina700.c

```diff
@@ -0,0 +1,333 @@
+{
+	int rc = 0;
+	struct ina700_data *data = dev->data;
+
+	ina700_lock_data(dev);
+
+	if ((chan == SENSOR_CHAN_VOLTAGE) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_16(dev, INA700_REG_VBUS, &data->voltage);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+	if ((chan == SENSOR_CHAN_CURRENT) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_16(dev, INA700_REG_CURRENT, &data->current);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+	if ((chan == SENSOR_CHAN_POWER) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_24(dev, INA700_REG_POWER, &data->power);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+	if ((chan == SENSOR_CHAN_DIE_TEMP) || (chan == SENSOR_CHAN_ALL)) {
+		rc = reg_read_16(dev, INA700_REG_DIE_TEMPERATURE, &data->temperature);
+
+		if (rc < 0) {
+			goto exit;
+		}
+	}
+
+exit:
+	ina700_unlock_data(dev);
+	return 0;
+}
+
+static int ina700_channel_get(const struct device *dev, enum sensor_channel chan,
+			      struct sensor_value *val)
+{
+	int rc = 0;
+	int32_t temp = 0;
+	struct ina700_data *data = dev->data;
+
+	ina700_lock_data(dev);
+
+	switch (chan) {
+	case SENSOR_CHAN_VOLTAGE:
+		temp = data->voltage * INA700_VOLTAGE_LSB;
+		val->val1 = temp / 1000000U;
+		val->val2 = temp % 1000000U;
+		break;
+
+	case SENSOR_CHAN_CURRENT:
+		temp = data->current * INA700_CURRENT_LSB;
+		val->val1 = temp / 1000000L;
+		val->val2 = temp % 1000000L;
+		break;
+
+	case SENSOR_CHAN_POWER:
+		temp = data->power * INA700_POWER_LSB;
+		val->val1 = temp / 1000000U;
+		val->val2 = temp % 1000000U;
+		break;
+
+	case SENSOR_CHAN_DIE_TEMP:
+		temp = FIELD_GET(IAN700_DIE_TEMPERATURE, data->temperature);
+		temp *= INA700_TEMPERATURE_LSB; /* m°C */
+		val->val1 = temp / 1000L;
+		val->val2 = temp % 1000L;
+		break;
+
+	default:
+		rc = -ENOTSUP;
+		break;
+	}
+
+	ina700_unlock_data(dev);
+
+	return 0;
+}
+
+#ifdef CONFIG_INA700_TRIGGER
+
+static void ina700_lock_data(const struct device *dev)
+{
+	struct ina700_data *data = dev->data;
+
+	k_mutex_lock(&data->mutex, K_FOREVER);
+}
+
+static void ina700_unlock_data(const struct device *dev)
+{
+	struct ina700_data *data = dev->data;
+
+	k_mutex_unlock(&data->mutex);
+}
+
+static void ina700_gpio_callback(const struct device *port, struct gpio_callback *cb, uint32_t pins)
+{
+	ARG_UNUSED(port);
+	ARG_UNUSED(pins);
+
+	struct ina700_data *data = CONTAINER_OF(cb, struct ina700_data, gpio_cb);
+
+	k_work_submit(&data->work);
+}
+
+static void ina700_work_cb(struct k_work *work)
+{
+	uint16_t status = 0;
+	struct ina700_data *data = CONTAINER_OF(work, struct ina700_data, work);
+
+	ina700_lock_data(data->this);
+
+	if (reg_read_16(data->this, INA700_REG_ALERT_DIAGNOSTICS, &status) == 0) {
+		if (status & INA700_ALERT_DIAG_CONVERSION_READY) {
+			if (ina700_sample_fetch(data->this, SENSOR_CHAN_ALL) == 0) {
+				if (data->handler_alert) {
+					data->handler_alert(data->this, data->trigger_alert);
+				}
+			}
+		}
+	}
+
+	ina700_unlock_data(data->this);
```

Comment:
flatten out nested-ifs :)
```suggestion
	if (reg_read_16(data->this, INA700_REG_ALERT_DIAGNOSTICS, &status) != 0) {
	        goto unlock;
	}    
	if (!(status & INA700_ALERT_DIAG_CONVERSION_READY)) {
	        goto unlock;
	}    
	if (ina700_sample_fetch(data->this, SENSOR_CHAN_ALL) != 0) {
	        goto unlock;
	}    
	if (data->handler_alert) {
		data->handler_alert(data->this, data->trigger_alert);
	}
unlock:
	ina700_unlock_data(data->this);
```